### ISSUE TITLE
[Atom 1.17.0] Missing configuration option when right clicking project folder or specific file

### ISSUE THREAD

**Issue by McNuggetSauce** [yongkangchen/remote-sync#388]
Remote Sync option is missing in the context menu when right-clicking a main project folder or specific file. It's available prior to Atom 1.17.0.

**Comment by McNuggetSauce** [comment]
Okay so I digged into this a little bit. It's caused by the non-existing class name selector for the context menu. 

Selector used is `.tree-view.full-menu` but the menu container only now has `tool-panel` and `tree-view`. Adding the `full-menu` back using the developer tool of Atom in `<div class="tool-panel tree-view full-menu" tabindex="-1" style="">` brings back the context menu item.

**Comment by dapeyo** [comment]
Hi

I have the same issue. Sub-menu to upload/download distant files is missing.

**Comment by gi-jones** [comment]
I confirm this problem too. Worked yesterday in 1.16. Gone now.

**Comment by cingwun0817** [comment]
I confirm this problem too. QQ

**Comment by cingwun0817** [comment]
ATOM 1.17.0  BOOM !!!

**Comment by dapeyo** [comment]
Sub Menu is still present inside the texteditor (right clic in it)

**Comment by lukwam** [comment]
Same issue here.

**Comment by timhecker** [comment]
+1

**Comment by ebro** [comment]
+1

**Comment by JordanPeltier** [comment]
Same problem here :'(

**Comment by tomaszh90** [comment]
+1

**Comment by roke22** [comment]
+1

**Comment by dtsdev** [comment]
@McNuggetSauce solution works like a charm. To fix issue "permanently" before it comes in update, open  `context.cson` (.atom/packages/remote-sync/menus) and change the following lines:
- line 2: from  `'atom-text-editor, .tree-view.full-menu .file':[` to `'atom-text-editor, .tool-panel.tree-view .file':[` 
- line 13: from `'.tree-view.full-menu .header.list-item':[` to `'.tool-panel.tree-view .header.list-item':[`
Restart Atom, done.

**Comment by dapeyo** [comment]
thanks for the hotfix

**Comment by alexandremmp** [comment]
Thanks @dtsdev ;)

**Comment by sorinpohontu** [comment]
Great, thanks @dtsdev :)

**Comment by Seykotron** [comment]
Thanks @McNuggetSauce it work like a charm! :) 👍

**Comment by JordanPeltier** [comment]
Nice, thanks @dtsdev :)

**Comment by webstrokepl** [comment]
Good job @dtsdev ;)

**Comment by dtsdev** [comment]
Thanks guys, but I'm merely dude with Ctrl+C/Ctrl+V abilities, so I'm rerouting all these thanks to @McNuggetSauce whose digging into the issue in the first place made this possible. Cheers, man :beer:

**Comment by nabu-3** [comment]
Dear dtsdev, thank's for this workarround.

**Comment by electricarts** [comment]
Thanks @McNuggetSauce! Fix works like a charm!

**Comment by nabu-3** [comment]
Dears,
This morning I've opened my Atom and the patch made last week was dissapeared. I've made the changes again.

**Comment by hiiwave** [comment]
Great! thanks @dtsdev

**Comment by nabu-3** [comment]
Dears,
This morning I've opened my Atom and the patch made last week was dissapeared. I've made the changes again.

**Comment by ZemanekRadek** [comment]
+1 thanks!

**Comment by hbulgarini** [comment]
Hello , 

I have the same issue and i have edited the file context.cson (.atom/packages/remote-sync/menus) but i still don't see the options once i do the right click. 
Here is how i have edited the context.cson for the remote-sync:

```
context-menu':
  'atom-text-editor, .tool-panel.tree-view .file':[
    label: 'Remote Sync',
    submenu:[
      {label: 'Upload File', command: 'remote-sync:upload-file'}
      {label: 'Download File', command: 'remote-sync:download-file'}
      {label: 'Diff File', command: 'remote-sync:diff-file'}
      {label: 'Delete File', command: 'remote-sync:delete-file'}
      {label: 'Monitor File (Upload on External Save)', command: 'remote-sync:monitor-file'}
    ]
  ]

> 

'.tool-panel.tree-view .header.list-item':[
    label: 'Remote Sync',
    submenu:[
      {label: 'Configure', command: 'remote-sync:configure'}
      {label: 'Upload Folder', command: 'remote-sync:upload-folder'}
      {label: 'Upload Folder Change', command: 'remote-sync:upload-git-change'}
      {label: 'Download Folder', command: 'remote-sync:download-folder'}
      {label: 'Diff Folder', command: 'remote-sync:diff-folder'}
      {label: 'Delete Folder', command: 'remote-sync:delete-folder'}
    ]
  ]
```

Is there anything i'm missing from this hotfix?

**Comment by linuxbastard** [comment]
This workaround no longer works for 1.17.2, new install

**Comment by williamsEye** [comment]
I tried the fix but only see the remote-sync context menu on the edit side.

**Comment by mikila** [comment]
+1 Thanks @dtsdev

**Comment by tcg** [comment]
Confirming that the change suggested at https://github.com/yongkangchen/remote-sync/issues/388#issuecomment-302121984 works for me, in `Atom 1.17.2`. 

For completeness, here's the content of my "patched" `.../remote-sync/menus/context.cson` file: 

```
'context-menu':
  'atom-text-editor, .tool-panel.tree-view .file':[
    label: 'Remote Sync',
    submenu:[
      {label: 'Upload File', command: 'remote-sync:upload-file'}
      {label: 'Download File', command: 'remote-sync:download-file'}
      {label: 'Diff File', command: 'remote-sync:diff-file'}
      {label: 'Delete File', command: 'remote-sync:delete-file'}
      {label: 'Monitor File (Upload on External Save)', command: 'remote-sync:monitor-file'}
    ]
  ]

  '.tool-panel.tree-view .header.list-item':[
    label: 'Remote Sync',
    submenu:[
      {label: 'Configure', command: 'remote-sync:configure'}
      {label: 'Upload Folder', command: 'remote-sync:upload-folder'}
      {label: 'Upload Folder Change', command: 'remote-sync:upload-git-change'}
      {label: 'Download Folder', command: 'remote-sync:download-folder'}
      {label: 'Diff Folder', command: 'remote-sync:diff-folder'}
      {label: 'Delete Folder', command: 'remote-sync:delete-folder'}
    ]
  ]
```

And, if you're not sure how to find that file (as I was, at first) try this: 

- View your installed packages list, and click the `remote-sync` package to view its details. 
- Below the package description, but *above* the **Settings** section, you should see a row of buttons: "View on Atom.io", "Report Issue", "LICENSE", and "View Code". Click "View Code". This should open a new Atom editor, including the folder + contents of the `remote-sync` package. 
- From there, it should be straightforward to find `/remote-sync/menus/context.cson`.

**Comment by nickv2002** [comment]
Works for me too.  Why has this not been patched in to the main branch and updated for all users yet?

**Comment by vikvarg** [comment]
After the patch you suggested now i see the menu after the right click, but if then i click on 'Edit Synk Config' nothing happens.

**Comment by yerfinojul** [comment]
Same as @vikvarg it does nothing when clicking on "Edit Sync Config"

**Comment by DISKONEKTeD** [comment]
Same. You can just manually do it apparently though:

Manually
    Add a file named .remote-sync.json to your project
    Add/configure with one of the contents below
    Save the file

and then add this (obviously modified to your needs):

```
{
  "transport": "scp",
  "hostname": "10.10.10.10",
  "port": 22,
  "username": "vagrant",
  "password": "vagrant",
  "keyfile": "/home/vagrant/.ssh/aws.pem",
  "passphrase": "your_passphrase",
  "target": "/home/vagrant/dirname/subdirname",
  "ignore": [
    ".remote-sync.json",
    ".git/**"
  ],
  "watch":[
    "/css/styles.css",
    "/index.html"
  ]
}
```

**Comment by DISKONEKTeD** [comment]
Actually, i got it to work just now, on the right hand corner, there was a red error (of atom) and so i clicked it. It opened a new tab: It said something about a file being from a previous build of atom and thus it was not compatible. So I chose to rebuild option (at the top right corner), and it did so successfully. Then I clicked on the reload atom option (again, top right corner). Once it reloaded I then proceed with the right click config on the project folder and the missing menu finally appeared.

**Comment by timothymarois** [comment]
This needs to be patched up already!

**Comment by AdsonCicilioti** [comment]
Also does not appear in the Packages menu.

Atom 1.18
OS: W10

**Comment by skrzepij** [comment]
In Atom 1.19 this problem still occurs.

**Comment by DragonAxe** [comment]
Problem still persists in Atom 1.21.0

**Comment by EvHaus** [comment]
For those blocked by this issue, I recommend switching to https://atom.io/packages/remote-sync-pro which is a fork of this project but with this (and other) issues fixed.

**Comment by daverickdunn** [comment]
The missing context menu has been patched and is working with Atom 1.21 and Atom IDE (1.22...?): https://github.com/yongkangchen/remote-sync/commit/3432d5287fa035dea822e7a7599bab62454cdf69

**Comment by hansnans** [comment]
fix from @McNuggetSauce works fine. Just add the .tool-panel.tree-view selector to the two blocks and its done
